service/src/integrationTest/kotlin/fi/espoo/evaka/ChildBuilder.kt

```diff
@@ -0,0 +1,280 @@
+package fi.espoo.evaka
+
+import fi.espoo.evaka.application.ApplicationStatus
+import fi.espoo.evaka.daycare.service.AbsenceType
+import fi.espoo.evaka.daycare.service.CareType
+import fi.espoo.evaka.placement.PlacementType
+import fi.espoo.evaka.shared.db.Database
+import fi.espoo.evaka.shared.dev.DevAssistanceNeed
+import fi.espoo.evaka.shared.dev.DevChild
+import fi.espoo.evaka.shared.dev.DevPerson
+import fi.espoo.evaka.shared.dev.insertTestAbsence
+import fi.espoo.evaka.shared.dev.insertTestApplication
+import fi.espoo.evaka.shared.dev.insertTestAssistanceNeed
+import fi.espoo.evaka.shared.dev.insertTestBackUpCare
+import fi.espoo.evaka.shared.dev.insertTestChild
+import fi.espoo.evaka.shared.dev.insertTestDaycareGroupPlacement
+import fi.espoo.evaka.shared.dev.insertTestNewServiceNeed
+import fi.espoo.evaka.shared.dev.insertTestPerson
+import fi.espoo.evaka.shared.dev.insertTestPlacement
+import fi.espoo.evaka.shared.dev.insertTestPlacementPlan
+import fi.espoo.evaka.shared.domain.FiniteDateRange
+import java.lang.IllegalStateException
+import java.time.LocalDate
+import java.util.UUID
+
+class ChildBuilder(
+    private val tx: Database.Transaction,
+    private val today: LocalDate
+) {
+    fun childOfAge(years: Int, months: Int = 0, days: Int = 0): ChildFixture {
+        val dob = today.minusYears(years.toLong()).minusMonths(months.toLong()).minusDays(days.toLong())
+        val childId = tx.insertTestPerson(DevPerson(dateOfBirth = dob))
+        tx.insertTestChild(DevChild(childId))
+        return ChildFixture(tx, today, childId)
+    }
+
+    class ChildFixture(
+        private val tx: Database.Transaction,
+        private val today: LocalDate,
+        val childId: UUID
+    ) {
+        fun hasAssistanceNeed() = AssistanceNeedBuilder(tx, today, this)
+        fun hasPlacementPlan() = PlacementPlanBuilder(tx, today, this)
+        fun hasBackupCare() = BackupCareBuilder(tx, today, this)
+        fun hasPlacement() = PlacementBuilder(tx, today, this)
+        fun hasAbsence() = AbsenceBuilder(tx, today, this)
+    }
+
+    class AssistanceNeedBuilder(
+        private val tx: Database.Transaction,
+        private val today: LocalDate,
+        private val childFixture: ChildFixture
+    ) {
+        private var employeeId: UUID? = null
+        private var factor: Double = 1.0
+        private var from: Int = 0
+        private var to: Int = 0
+
+        fun withFactor(factor: Double) = this.apply { this.factor = factor }
+        fun fromDay(day: Int) = this.apply { this.from = day }
+        fun toDay(day: Int) = this.apply { this.to = day }
+        fun createdBy(employeeId: UUID) = this.apply { this.employeeId = employeeId }
+
+        fun execAndReturn(): ChildFixture {
```

Comment:
I think `save`, `persist` or something like that would describe the intent better then `exec`. I also think that the `AndReturn` part is kind of redundant as well.

Reply:
Yeah +1 for `save` over `exec`. 

For placements the return part is important because you may want to add some stuff related to the placement before returning
```
.addPlacement() ... .save()
   .withServiceNeed() ... .saveAndReturn()
   .withGroupPlacement() ... .saveAndReturn()
  .return()
.addPlacement() ...
```
In most cases there is no next level though so saving without returning is not necessary. My thought was that this way the API is nevertheless more consistent, but maybe there's some better option.. Maybe `save` should return by default since that is most common and then have some other function name for the special case of `save and then add some child entities to this`.